### PR TITLE
New feature: conformation request before overwriting differing rdf model

### PR DESCRIPTION
**Pull request by Shamanou**
This feature allows the user to reconsider before overwriting his/her RDF model with a shared RDF model. The dialog box does not appear if the loaded and exciting models are equal. If the dialog is canceled the user is returned to the rdf-model-list-dialog. If the user presses ok the user is returned to the rdf model editor and the rdf model is overwritten.

Edits are in big blocks because of style edits. 
Also some commits to add badges.
Commits contain more specific information.

### CODE REVIEW

File: extensions/fairifier-rdf-extension/dtls/fairifier/LoadRdfSkeletonCommand.java [hunk 1]
```diff
@@ -49,11 +47,13 @@ public void doPost(HttpServletRequest req, HttpServletResponse res)
...
-            throws Exception {
-        Model out = rdfSkeletonService.loadModel(request.getParameter("projectId")); // this is the project where the
-                                                                                     // skeleton comes from
+    private void performProcess(HttpServletRequest request, HttpServletResponse response,
+            Project project, Process process) throws Exception {
+        Model out = rdfSkeletonService.loadModel(request.getParameter("projectId")); // this is the
+                                                                                     // project
+                                                                                     // where the
+                                                                                     // skeleton
+                                                                                     // comes from
```
Comment (kburger) [inline extensions/fairifier-rdf-extension/dtls/fairifier/LoadRdfSkeletonCommand.java:56]: This comment style is not preferred, and the reformatted version is a perfect example of why. Having one to two words per line in a comment suffix/trailing style is not very readable and wastes a lot of horizontal and vertical whitespace. For inline comments on constants/variable initialization it is acceptable (see [here](https://google.github.io/styleguide/javaguide.html#s2.3.3-non-ascii-characters) for good examples). For this case, I'd say put the comment on the line above.

File: extensions/fairifier-rdf-extension/scripts/rdf-schema-alignment.js [hunk 2]
```diff
@@ -1,335 +1,343 @@
...
+
+  RdfSchemaAlignment._defaultNamespace = this._schema.baseUri;
+
+  // initialize vocabularyManager
+  this._prefixesManager = new RdfPrefixesManager(this, this._schema.prefixes);
+
+  this
+          ._replaceBaseUri(RdfSchemaAlignment._defaultNamespace
+                  || window.location.protocol + "//" + window.location.host
+                  + "/", true);
```
Comment (kburger) [inline extensions/fairifier-rdf-extension/scripts/rdf-schema-alignment.js]: What's with the whitespace between `this` and `._replaceBaseUri(`?

File: extensions/fairifier-rdf-extension/scripts/rdf-schema-alignment.js [hunk 3]
```diff
@@ -1,335 +1,343 @@
...
-                }
-            );
-    }).appendTo(footer);
-    
-    $('<button></button>').addClass('button').text("Cancel").click(function() {
-        DialogSystem.dismissUntil(self._level - 1);
-    }).appendTo(footer);
+  var self = this;
+
+  $('<button></button>').addClass('button').html("&nbsp;&nbsp;OK&nbsp;&nbsp;")
```
Comment (kburger) [inline extensions/fairifier-rdf-extension/scripts/rdf-schema-alignment.js:50]: Not something for now, but for the future: padding with `&nbsp;` could probably be replaced with css margin/padding.

File: extensions/fairifier-rdf-extension/scripts/rdf-schema-alignment.js [hunk 4]
```diff
@@ -1,335 +1,343 @@
...
-    elmts.editBaseUriLink.click(function(evt){
-    	evt.preventDefault();
-    	self._editBaseUri($(evt.target));
-    });
+  var self = this;
+  $(
+          '<p>'
+                  + 'The Semantic model below specifies how the RDF data that will get generated from your grid-shaped data. '
+                  + 'The cells in each record of your data will get placed into nodes within the model. Configure the model by specifying which column to substitute into which node.'
+                  + '</p>').appendTo(body);
```
Comment (kburger) [inline extensions/fairifier-rdf-extension/scripts/rdf-schema-alignment.js]: What's with the whitespace between `$(`, `'<p>'`, and the rest of the text?

**Review by kburger — APPROVED** [review]
It's kind of hard to see the actual changes between the style/formatting updates and the logic updates. Maybe split them up in different commits. For the javascript itself, I don't have much feedback on it at the moment. For the java code, minor comments.

File: extensions/fairifier-rdf-extension/scripts/rdf-schema-alignment.js (hunk 2 above)
Comment (Shamanou) [reply extensions/fairifier-rdf-extension/scripts/rdf-schema-alignment.js]: This was a result of style formatting, solved by shortening the sentence by introducing new variable

**Pull request merged**